app/utilities/schema.py

```diff
@@ -206,8 +235,12 @@ def load_schema_from_url(
         language_code=language_code,
         pid=pid,
     )
+    if cir_instrument_id:
```

Comment:
Only minor, but we could keep this method generic if we set the parameters in the calling methods?

Reply:
Yeah, I would prefer generic, my only holding back was trying to keep `test_load_schema_from_url` as is and if parameters as a dictionary is passed in its not hashable for the lru cache, and if I encode the params, I might as well join them up in the caller too - works with an Immutable dictionary though, is what I just pushed up what you had in mind? I had to add some param args in the tests, but outside tests I think this works

Another option is to have `**kwargs` and use this to form the parameters, but I don't like that because it means the existing implementation is then called like this which looks really odd
`load_schema_from_url(schema_url=url, language_code="en", language="en")` 

Or I could go ahead and pass in the constructed url to the function. Gone with the dict but on the fence with which is better, so all for opinions

Reply:
Yeah I see what you mean 🤔. What you've pushed is what I had in mind and it looks OK to me now, so happy with it as is 👍

Reply:
Whole url now goes in instead so still generic but this issue avoided